riva/proto/riva_nmt.proto

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+//
+// NVIDIA CORPORATION and its licensors retain all intellectual property
+// and proprietary rights in and to this software, related documentation
+// and any modifications thereto.  Any use, reproduction, disclosure or
+// distribution of this software and related documentation without an express
+// license agreement from NVIDIA CORPORATION is strictly prohibited.
+
+syntax="proto3";
+
+package nvidia.riva.nmt;
+
+option cc_enable_arenas = true;
+option go_package = "nvidia.com/riva_speech";
+
+/*
+ *  RivaTranslation service provides rpcs to translate between languages.
+ *
+ */
+service RivaTranslation {
+
+  // Translate text to text, from a source to a target language.  Currently source language is required, along with the model name.
+  // Multiple texts may be passed per request up to the given batch size for the model, which is set at translation pipeline creation time.
+rpc TranslateText(TranslateTextRequest) returns (TranslateTextResponse) {}
+
+  // Lists the available language pairs and models names to be used for TranslateText
+  rpc ListSupportedLanguagePairs(AvailableLanguageRequest) returns (AvailableLanguageResponse) {}
+}
+
+// request for synchronous translation of each text in texts.
+// Available languages can be queries using ListSupportLanguagePairs RPC.
+// source and target languages must be specified, are currently two character ISO codes, this will likely change to BCP-47 inline with other Riva Services for GA.
+//
+message TranslateTextRequest {
+    repeated string texts = 1;
+    string model = 2;
+    string source_language = 3;
+    string target_language = 4;
```

Comment:
can it support multiple target languages presently ?

Reply:
No, not per request, but multilingual models can support multiple targets

Reply:
single request can only take one src, one target lang as an input ? and to get multilingual translation multiple requests are required ?

Reply:
Assuming for multilingual translation , multiple requests are to be made, wouldn't it be better if TranslateTextResponse just has one Translation and not the list ? existing format of TranslateTextResponse  to contain a list of Translation, is confusing from the request:response pov since clearly it is 1:1 and not 1:many as anyways there will be multiple requests anyways ? Can you please clarify this ?

Reply:
We support batching in NMT like other services, and to create a batch you pass in multiple texts.

Reply:
Does batching support multilingual translation( multiple texts be of different languages )? If not , then Translation should look like below: message Translation{
string text =1;} and message TranslateTextResponse {
repeated Translation translations =1;
string language =2;} Can you clarify ?

Reply:
That is is also an option sure.  If you feel strongly we should look at this the next release can you open a jira to track?  Please keep in mind this proto has already gone out and we cannot change it for this release.

Reply:
I think the proto architecture should be as clear and concise as possible. Let me create a JIRA and revisit this for the next release. Thanks!